camel/configs/base_config.py

```diff
@@ -14,9 +14,19 @@
 from __future__ import annotations
 
 from abc import ABC
-from dataclasses import dataclass
+from typing import Any
 
+from pydantic import BaseModel, ConfigDict
 
-@dataclass(frozen=True)
-class BaseConfig(ABC):  # noqa: B024
-    pass
+
+class BaseConfig(ABC, BaseModel):
+    model_config = ConfigDict(
+        arbitrary_types_allowed=True,
+        extra="forbid",
+        frozen=True,
+        # UserWarning: conflict with protected namespace "model_"
+        protected_namespaces=(),
+    )
+
+    def as_dict(self) -> dict[str, Any]:
```

Comment:
@onemquan @ZackYule I see that as_dict() and model_dump() are equivalent, maybe they can be unified? I think as_dict() is pretty good, what do you think?